builders/ethereum/dev-env/thirdweb.md

```diff
@@ -110,184 +112,180 @@ thirdweb offers SDKs for a range of programming languages, such as React, React
-      Web3Button,
-    } from '@thirdweb-dev/react';
-
-    // Your smart contract address
-    const contractAddress = 'INSERT_CONTRACT_ADDRESS';
-
-    function App() {
-      const { contract } = useContract(contractAddress);
-      const { mutateAsync, isLoading, error } = useContractWrite(
-        contract,
-        'INSERT_NAME'
-      );
-
-      return (
-        <Web3Button
-          contractAddress={contractAddress}
-          // Calls the 'INSERT_NAME' function on your smart contract
-          // with 'INSERT_ARGUMENT' as the first argument
-          action={() => mutateAsync({ args: ['INSERT_ARGUMENT'] })}
-        >
-          Send Transaction
-        </Web3Button>
-      );
-    }
-    ```
+const number = await readContract({
+      contract: contract,
+      method: 'number',
+      params: [],
+    });
+```
 
 ### Connect Wallet {: #connect-wallet }
 
-There are a couple of ways that you can create a custom [connect wallet](https://portal.thirdweb.com/react/connecting-wallets){target=\_blank} experience. You can use the [`ConnectWallet` component](https://portal.thirdweb.com/react/connecting-wallets#using-the-connect-wallet-button){target=\_blank} or, for a more customizable approach, you can use the [`useConnect` hook](https://portal.thirdweb.com/react/connecting-wallets#using-hooks){target=\_blank}.
-
-The following example will show you how to use the `ConnectWallet` component. To go this route, you will need to specify the supported wallets and pass them to your provider.
-
-```javascript
-import {
-  ThirdwebProvider,
-  metamaskWallet,
-  coinbaseWallet,
-  walletConnectV1,
-  walletConnect,
-  safeWallet,
-  paperWallet,
-} from '@thirdweb-dev/react';
-
-function MyApp() {
-  return (
-    <ThirdwebProvider
-      supportedWallets={[
-        metamaskWallet(),
-        coinbaseWallet(),
-        walletConnect({
-          projectId: 'INSERT_YOUR_PROJECT_ID', // optional
-        }),
-        walletConnectV1(),
-        safeWallet(),
-        paperWallet({
-          clientId: 'INSERT_YOUR_CLIENT_ID', // required
-        }),
-      ]}
-      activeChain={Moonbeam}
-    >
-      <App />
-    </ThirdwebProvider>
-  );
-}
-```
+Next, you can customize the [Connect Button](https://portal.thirdweb.com/typescript/v5/react/components/ConnectButton){target=\_blank} to tailor it our desired wallets. The Connect Button accepts an optional  `wallets` parameter with an array of wallets. You can add or remove wallets from the wallets array to change the options available to users. ThirdWeb also offers a [ConnectButton Playground](https://thirdweb.com/dashboard/connect/playground) to customize and view changes in real-time given the high degree of flexibility offered by the button. The below snippet will also include logic to display the counter value and the increment button.
 
-Next, you'll need to add in a connect wallet button to prompt end-users to login with any of the above supported wallets.
+```typescript title="App.tsx"
```

Comment:
This will likely sound silly to someone as experienced as yourself, but I will share it anyway. 

As a greener developer, your import statements in these code snippets freak me out. They keep changing from step to step. Some things get imported at the top in one step, and then they aren't there anymore in the next step. That is enough to make a junior developer wonder if they are putting everything in the right place. Sharing the completed code at the end is helpful but I almost missed that it was there at the end. Adding a mention and a link near the top of the Create Application section would be a big help for less experienced devs to draw attention to the complete sample.